lidro/create_virtual_point/pointcloud/convert_geodataframe_to_las.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+""" this function convert GeoDataframe "virtual points" to LIDAR points
+"""
+import logging
+import os
+
+import geopandas as gpd
+import laspy
+import numpy as np
+import pandas as pd
+
+
+def geodataframe_to_las(virtual_points: gpd.GeoDataFrame, output_dir: str, crs: str):
+    """This function convert virtual points (GeoDataframe) to LIDAR points with classification (66) for virtual points
+
+    Args:
+        virtual_points (gpd.GeoDataFrame): A GeoDataFrame containing virtuals points from Mask Hydro
+        output_dir (str): folder to output LAS
+        crs (str): a pyproj CRS object used to create the output GeoJSON file
+    """
+    # Combine all virtual points into a single GeoDataFrame
+    grid_with_z = gpd.GeoDataFrame(pd.concat(virtual_points, ignore_index=True))
+
+    # Extract x, y, and z coordinates
+    grid_with_z["x"] = grid_with_z.geometry.x
+    grid_with_z["y"] = grid_with_z.geometry.y
+    grid_with_z["z"] = grid_with_z.geometry.z
+
+    # Create a DataFrame with the necessary columns
+    las_data = pd.DataFrame(
+        {
+            "x": grid_with_z["x"],
+            "y": grid_with_z["y"],
+            "z": grid_with_z["z"],
+            "classification": np.full(len(grid_with_z), 68, dtype=np.uint8),  # Add classification of 68
+        }
+    )
+
+    # Create a LAS file with laspy
+    header = laspy.LasHeader(point_format=6, version="1.4")
+
+    # Create a LAS file with laspy and add the VLR for CRS
+    las = laspy.LasData(header)
+    las.x = las_data["x"].values
+    las.y = las_data["y"].values
+    las.z = las_data["z"].values
+    las.classification = las_data["classification"].values
```

Comment:
j'ai l'impression que tu peux supprimer des étapes intermédiaires vu que ton entrée est déjà une GeoDataFrame (tu fais déjà la concaténation en dehors si j'ai bien compris) : 
[ça vaut le coup de revérifier ce que j'ai écrit ici, c'est juste un brouillon]

```suggestion
def geodataframe_to_las(virtual_points: gpd.GeoDataFrame, output_dir: str, crs: str):
    """This function convert virtual points (GeoDataframe) to LIDAR points with classification (66) for virtual points

    Args:
        virtual_points (gpd.GeoDataFrame): A GeoDataFrame containing virtuals points from Mask Hydro
        output_dir (str): folder to output LAS
        crs (str): a pyproj CRS object used to create the output GeoJSON file
    """

    # Create a LAS file with laspy
    header = laspy.LasHeader(point_format=6, version="1.4")

    # Create a LAS file with laspy and add the VLR for CRS
    las = laspy.LasData(header)
    las.x = grid_with_z.geometry.x.values
    las.y =grid_with_z.geometry.y.values
    las.z = grid_with_z.geometry.z.values
    las.classification = np.full(len(grid_with_z), 68, dtype=np.uint8),  # Add classification of 68
```

Reply:
review : [b62bc00](https://github.com/IGNF/lidro/pull/10/commits/b62bc006b89e62d24708c39c2af3b8f525d2f7ce)

Reply:
proposition : garder geodataframe + 
```python 
def geodataframe_to_las(virtual_points: List, output_dir: str, crs: str, classes: int):
    """This function convert virtual points (GeoDataframe) to LIDAR points with classification for virtual points

    Args:
        virtual_points (List): A list containing virtuals points by hydrological entity
        output_dir (str): folder to output LAS
        crs (str): a pyproj CRS object used to create the output GeoJSON file
        classes (int): The number of the classe assign those virtual points
    """
    # Combine all virtual points into a single GeoDataFrame
    grid_with_z = pd.concat(virtual_points, ignore_index=True)

    # Create a LAS file with laspy
    header = laspy.LasHeader(point_format=6, version="1.4")

    # Create a LAS file with laspy and add the VLR for CRS
    las = laspy.LasData(header)
    las.x = grid_with_z.geometry.x
    las.y = grid_with_z.geometry.y
    las.z = grid_with_z.geometry.z
    las.classification = classes * np.ones(len(grid_with_z.index))

    # Add CRS information to the VLR
    vlr = laspy.vlrs.known.WktCoordinateSystemVlr(crs.to_wkt())
    las.vlrs.append(vlr)

    # Save the LAS file
    output_laz = os.path.join(output_dir, "virtual_points.laz")
    with laspy.open(output_laz, mode="w", header=las.header, do_compress=True) as writer:
        writer.write_points(las.points)

    logging.info(f"Virtual points LAS file saved to {output_laz}")
```